flight/PiOS/Common/pios_hal.c

```diff
@@ -1197,3 +1229,138 @@ void PIOS_HAL_ConfigureSerialSpeed(uintptr_t com_id,
 			break;
 	}
 }
+
+#ifdef PIOS_INCLUDE_I2C
+static int PIOS_HAL_ConfigureI2C(uint32_t *id,
+		const struct pios_i2c_adapter_cfg *cfg) {
+	if (!*id) {
+		// Not already initialized.
+		if (PIOS_I2C_Init(id, cfg)) {
+			PIOS_DEBUG_Assert(0);
+			AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
+
+			return -1;
+		}
+	}
+
+	if (PIOS_I2C_CheckClear(*id) != 0) {
+		AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
+		return -2;
+	}
+
+	if (AlarmsGet(SYSTEMALARMS_ALARM_I2C) ==
+			SYSTEMALARMS_ALARM_UNINITIALISED) {
+		AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_OK);
+	}
+
+	PIOS_WDG_Clear();
+
+	return 0;
+}
+#endif // PIOS_INCLUDE_I2C
+
+int PIOS_HAL_ConfigureExternalBaro(HwSharedExtBaroOptions baro,
+		uint32_t *i2c_id,
+		const struct pios_i2c_adapter_cfg *i2c_cfg)
+{
+#if !defined(PIOS_INCLUDE_I2C)
+	return -1;
+#else
+	if (baro == HWSHARED_EXTBARO_NONE) {
+		return 0;
+	}
+
+	int ret = PIOS_HAL_ConfigureI2C(i2c_id, i2c_cfg);
+
+	if (ret) goto done;
+
+	switch(baro) {
+#if defined(PIOS_INCLUDE_BMP280)
```

Comment:
Shouldn't this be below case? Otherwise if someone sets the option to BMP280 but `PIOS_INCLUDE_BMP280` is not defined it will crash.